.github/workflows/docs.yml

```diff
@@ -0,0 +1,56 @@
+name: "Release Docs"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'mkdocs/**'
+      - 'src/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+permissions:
+  contents: write
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
```

Comment:
Do we need all history of the repo for doc release? If not, the default 1 should be fine.

https://github.com/actions/checkout/blob/main/README.md?plain=1#L120